### ISSUE TITLE
For some users, Snap install does not have access to or prompt user to grant access to OS keyring

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#33576]
## Description 

After the update to v1.58.137, the snap version of the browser stopped saving any newly entered secure data (eg passwords). While old data was retained, anything new would only be remembered until the browser was closed. Note that no data was set to be cleared "on exit" here and this was tested on a fresh profile as well as a used one. 

User also stated that there was no `Brave safe storage` entry in the keyring (but not clear how the old data was retained then?) and was only able to ascertain that the snap version was at play here by uninstalling and reinstalling via apt. 

## Steps to Reproduce

   1. Have Ubuntu 22.04 LST and Snap version of Brave installed 
   2. Save login data for any site
   3. Close and relaunch the browser
   4. Observe that the saved data is not retained

## Brave version (brave://version info)

Version 1.58.137 Chromium: 117.0.5938.153 (Official Build) (64-bit) on Ubuntu 22.04 LST

## Version/Channel Information:

- Can you reproduce this issue with the current release? Y
- Can you reproduce this issue with the beta channel? NA
- Can you reproduce this issue with the nightly channel? NA

## Miscellaneous Information:

User report:
https://community.brave.com/t/brave-does-not-retain-login-info-and-password-once-close/509522

**Comment by rebron** [comment]
cc: @mihaiplesa @wknapik

**Comment by loimu** [comment]
It's a simple fix:
```
snap connect brave:password-manager-service
```
Of course, this rule can be embedded into snap package itself.

**Comment by loimu** [comment]
But as I realize Brave doesn't store any passwords there anymore, only encryption key from it's own safe storage.

**Comment by loimu** [comment]
@wknapik I'm using `--password-store=kwallet` option already. Without running `snap connect brave:password-manager-service` it does nothing.

**Comment by fmarier** [comment]
> So, as far as I can tell, this works on 1.65.114 on Linux x64. I assume the secrets are not supposed to show up in my KeePassXC vault as new entries and the browser just stores one secret in the secret service and encrypts/decrypts the individual passwords using that (that's just a guess, @fmarier please confirm).

Correct. The browser stores a single encryption key (used for saved passwords and cookies) in the keyring.

> AFAICT there is no issue here. Perhaps someone else can test with kwallet.

Your description above  (the 4 bullet points) is what I would expect to see.

It seems like the underlying problem here was the lack of `snap connect`. Maybe we should document this somewhere? We have a [support page for the password manager](https://support.brave.com/hc/en-us/articles/360018185951-How-do-I-use-the-built-in-password-manager), but since this is specific to the Snap, maybe it should be added to https://brave.com/linux/#snap or the package description on https://snapcraft.io/brave?

**Comment by bsclifton** [comment]
Closing as stale. If this is still happening, let's reopen the issue. Thanks!
